superset-frontend/src/components/TimezoneSelector/index.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect } from 'react';
+import moment from 'moment-timezone';
+
+import { NativeGraySelect as Select } from 'src/components/Select';
+import { useChangeEffect } from 'src/common/hooks/useChangeEffect';
+
+const DEFAULT_TIMEZONE = 'GMT Standard Time';
+
+const offsetsToName = {
+  '-300-240': ['Eastern Standard Time', 'Eastern Daylight Time'],
+  '-360-300': ['Central Standard Time', 'Central Daylight Time'],
+  '-420-360': ['Mountain Standard Time', 'Mountain Daylight Time'],
+  '-420-420': [
+    'Mountain Standard Time - Phoenix',
+    'Mountain Standard Time - Phoenix',
+  ],
+  '-480-420': ['Pacific Standard Time', 'Pacific Daylight Time'],
+  '-540-480': ['Alaska Standard Time', 'Alaska Daylight Time'],
+  '-600-600': ['Hawaii Standard Time', 'Hawaii Daylight Time'],
+  '60120': ['Central European Time', 'Central European Daylight Time'],
+  '00': [DEFAULT_TIMEZONE, DEFAULT_TIMEZONE],
+  '060': ['GMT Standard Time - London', 'British Summer Time'],
+};
+
+const currentDate = moment();
+const JANUARY = moment([2021, 1]);
+const JULY = moment([2021, 7]);
+
+const getOffsetKey = (name: string) =>
+  JANUARY.tz(name).utcOffset().toString() +
+  JULY.tz(name).utcOffset().toString();
+
+const getTimezoneName = (name: string) => {
+  const offsets = getOffsetKey(name);
+  return (
+    (currentDate.tz(name).isDST()
+      ? offsetsToName[offsets]?.[1]
+      : offsetsToName[offsets]?.[0]) || name
+  );
+};
+
+export interface TimezoneProps {
+  onTimezoneChange: (value: string) => void;
+  timezone?: string | null;
+}
+
+const ALL_ZONES = moment.tz
+  .countries()
+  .map(country => moment.tz.zonesForCountry(country, true))
+  .flat();
+
+const TIMEZONES: moment.MomentZoneOffset[] = [];
+ALL_ZONES.forEach(zone => {
+  if (
+    !TIMEZONES.find(
+      option => getOffsetKey(option.name) === getOffsetKey(zone.name),
+    )
+  ) {
+    TIMEZONES.push(zone); // dedupe zones by offsets
```

Comment:
This will remove some popular timezones and replace them with some more rare timezones, right?

Are there too many to leave the duplicates? Do you think we could sort by popularity (population) somehow?

Reply:
Yeah, I talked with @yousoph about this.. there were originally about 450 timezones. There's no sense of popularity, but what I did was rename the US duplicates into "popular" names, so we have PST/PDT, etc.. and we can continue to add more to that list. Moment.js also doesn't give you direct access to the zone name, just region, but when printing the region it's available, so I thought there may be a way to parse that out in maybe a v2.